bgpd/bgp_bmp.c

```diff
@@ -1329,6 +1329,33 @@ static int bmp_stats(struct thread *thread)
 	return 0;
 }
 
+/* read from the BMP socket to detect session termination */
+static int bmp_read(struct thread *t)
+{
+	struct bmp *bmp = THREAD_ARG(t);
+	char buf[1024];
+	ssize_t n;
+
+	bmp->t_read = NULL;
+
+	n = read(bmp->socket, buf, sizeof(buf));
+	if (n >= 1) {
+		zlog_info("bmp[%s]: unexpectedly received %zu bytes", bmp->remote, n);
```

Comment:
1. Isn't this a normal condition ? why is this "unexpected" ?
2. As bmp_wrerr dumps a log messages in both the error cases, do we need this log ?

Reply:
Hi @srimohans This log was added to confirm if it is an expected state or not, and in my testing this log has never triggered.
Currently testing on 32 FRR nodes, which have handled over 100M bmp messages so far, over the last 2 weeks, and everything is still stable (sessions were going stale after 24-48 hours previously).
So it appears this state is never reached, making it reasonable to say it's unexpected. But yes I agree that when N > 1, we don't really care.

Reply:
Thanks.